code/modules/projectiles/projectile/magic.dm

```diff
@@ -406,11 +406,21 @@
 	. = ..()
 	if(isliving(A) && locker_suck)
 		var/mob/living/M = A
-		if(M.anti_magic_check())			// no this doesn't check if ..() returned to phase through do I care no it's magic ain't gotta explain shit
-			M.visible_message("<span class='warning'>[src] vanishes on contact with [A]!</span>")
-			return PROJECTILE_DELETE_WITHOUT_HITTING
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] brutally hits [A]!</span>", "<span class='boldwarning'>You are hit by a a locker!")
+			var/hit_zone = pick(BODY_ZONE_HEAD,BODY_ZONE_CHEST,BODY_ZONE_PRECISE_GROIN) //hit in the balls by a locker
+			M.apply_damage(20, BRUTE, hit_zone, M.run_armor_check(hit_zone, MELEE))
+
+			//launch very hard
+			var/atom/throw_target = get_edge_target_turf(M, get_dir(src, get_step_away(M, src)))
+			M.safe_throw_at(throw_target, 300, 2, null, TRUE, FALSE, null, MOVE_FORCE_OVERPOWERING)
```

Comment:
This is entirely random and results in your target being flung into you at speeds faster than framerate. The combination of the absurd length of the throw range and randomness makes this extremely inconsistent. Might be related to the runtime I found.

Reply:
At close range it becomes really unpredictable because we’re using the projectiles location, longer range was quite a bit better. I didn’t have problems with them being flung fast at all. Also the range was meant to be 30, which is still absurd but it’s flying lockers and everyone loves launching people. 

I can cheese it a bit and use the firers location, it would mean the launch direction changes when the firer moves but no one would probably notice

Reply:
Firer is fine, very unlikely is it that you'd end up with a firer being opposite to the shooter upon projectile impact before it despawns.